data-prepper-plugins/peer-forwarder/README.md

```diff
@@ -1,20 +1,34 @@
 # Peer Forwarder
+This processor forwards `ExportTraceServiceRequest` via gRPC to other Data Prepper instances. The primary usecase of this processor is 
+to ensure that groups of traces are aggregated by trace ID and processed by the same Prepper instance.
 
-This is a processor that forwards `ExportTraceServiceRequest` through [grpc] towards the given peer Data-Prepper source Ip addresses. 
-It regroups spans by trace ID and forward the group to the corresponding peer by consistent hashing. 
+Presently peer discovery is provided by either a static list (configured in yaml) or by a DNS record lookup.
 
 ## Configuration
+Static list example:
+```
+processor:
+    - peer_forwarder:
+        time_out: 300
+        span_agg_count: 48
+        discovery_mode: "static"
+        static_endpoints:
+        - "data-prepper1"
+        - "data-prepper2"
+```
 
-Example configuration is as follows:
-
+DNS lookup example:
 ```
 processor:
     - peer_forwarder:
-       data_prepper_ips: ["70.162.32.1", "70.162.32.2"]
-       time_out: 300
-       span_agg_count: 48
+        time_out: 300
+        span_agg_count: 48
+        discovery_mode: "dns"
+        hostname_for_dns_lookup: "data-prepper-cluster"
 ```
 
-- `data_prepper_ips`: IP addresses for all running Data-Preppers including the localhost.
-- `time_out`: timeout in seconds for sending `ExportTraceServiceRequest`. Defaults to 300
-- `span_agg_count`: batch size for number of spans per `ExportTraceServiceRequest`. Defaults to 48.
\ No newline at end of file
+* `time_out`: timeout in seconds for sending `ExportTraceServiceRequest`. Defaults to 3 seconds.
+* `span_agg_count`: batch size for number of spans per `ExportTraceServiceRequest`. Defaults to 48.
+* `discovery_mode`: peer discovery mode to be used. Allowable values are `static` and `dns`. Defaults to `static`
+* `static_endpoints`: list containing endpoints of all Data Prepper instances
+* `hostname_for_dns_lookup`: single hostname to query DNS against. Typically used by creating multiple A Records for the same domain
```

Comment:
Nit: what does "multiple A Records" mean?

Reply:
I'll add a link in the readme to https://www.cloudflare.com/learning/dns/dns-records/dns-a-record/